ubireader/ubifs/decrypt.py

```diff
@@ -0,0 +1,98 @@
+import os
+import struct
+
+from ubireader.ubifs.defines import *
+from ubireader.debug import error
+from cryptography.hazmat.primitives.ciphers import (
+   Cipher, algorithms, modes
+)
+
+MASTER_KEY = b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
+IV = b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
+
+def set_master_key(key):
+    global MASTER_KEY
+    MASTER_KEY = key
```

Comment:
I would adapt the code not to use globals. It can have very nasty and implicit side effects.